harness/determined/common/experimental/experiment.py

```diff
@@ -115,11 +130,30 @@ def reload(self) -> None:
         resp = bindings.get_GetExperiment(self._session, experimentId=self.id).experiment
         self._hydrate(resp)
 
+    def set_name(self, name: str) -> None:
+        """Set (overwrite if existing) name on the experiment."""
+        req_body = bindings.v1PatchExperiment(id=self.id, name=name)
+        resp = bindings.patch_PatchExperiment(self._session, experiment_id=self.id, body=req_body)
+        self.name = resp.experiment.name if resp.experiment else None
+
+    def set_description(self, description: str) -> None:
+        """Set description (overwrite if existing) description on the experiment."""
+        req_body = bindings.v1PatchExperiment(id=self.id, description=description)
+        resp = bindings.patch_PatchExperiment(self._session, experiment_id=self.id, body=req_body)
+        self.description = resp.experiment.description if resp.experiment else None
+
+    def set_notes(self, notes: str) -> None:
+        """Set notes (overwrite if existing) description on the experiment."""
+        req_body = bindings.v1PatchExperiment(id=self.id, notes=notes)
+        resp = bindings.patch_PatchExperiment(self._session, experiment_id=self.id, body=req_body)
+        self.notes = resp.experiment.notes if resp.experiment else None
+
     def activate(self) -> None:
         bindings.post_ActivateExperiment(self._session, id=self._id)
 
     def archive(self) -> None:
         bindings.post_ArchiveExperiment(self._session, id=self._id)
+        self.archived = True
```

Comment:
In other files we have a test that this won't happen if the POST fails. I don't know if it needs a test per se. But there's something I like about how the existence of the test specifies the behavior.

If you don't want to copy/paste the tests from [test_model.py](https://github.com/determined-ai/determined/blob/main/harness/tests/determined/common/experimental/test_model.py#L106), then what do you think about putting it in a docstring?

Reply:
having tests for this seems like a good idea. done.